src/plugins/core/filters.ts

```diff
@@ -0,0 +1,343 @@
+        }
+      }
+    }
+  }
+
+  private createFilterTable(zone: Zone): FilterTable {
+    return new FilterTable(zone);
+  }
+
+  /** Extend a table down one row */
+  private extendTableDown(sheetId: UID, table: FilterTable) {
+    const newZone = { ...table.zone, bottom: table.zone.bottom + 1 };
+    this.history.update("tables", sheetId, table.id, "zone", newZone);
+    for (let filterIndex = 0; filterIndex < table.filters.length; filterIndex++) {
+      const filter = table.filters[filterIndex];
+      const newFilterZone = {
+        ...filter.zoneWithHeaders,
+        bottom: filter.zoneWithHeaders.bottom + 1,
+      };
+      this.history.update(
+        "tables",
+        sheetId,
+        table.id,
+        "filters",
+        filterIndex,
+        "zoneWithHeaders",
+        newFilterZone
+      );
+    }
+    return;
+  }
+
+  /**
+   * Check if an UpdateCell command should cause the given table to be extended by one row.
+   *
+   * The table should be extended if all of these conditions are true:
+   * 1) The updated cell is right below the table
+   * 2) The command adds a content to the cell
+   * 3) No cell right below the table had any content before the command
+   * 4) Extending the table down would not overlap with another filter
+   * 5) Extending the table down would not overlap with a merge
+   *
+   */
+  private canUpdateCellCmdExtendTable(
+    { content: newCellContent, sheetId, col, row }: UpdateCellCommand,
+    table: FilterTable
+  ) {
+    if (!newCellContent) {
+      return;
+    }
+
+    const zone = table.zone;
+    if (!(zone.bottom + 1 === row && col >= zone.left && col <= zone.right)) {
+      return false;
+    }
+
+    for (const col of range(zone.left, zone.right + 1)) {
+      // Since this plugin is loaded before CellPlugin, the getters still give us the old cell content
+      const cellContent = this.getters.getCell(sheetId, col, row)?.content;
+      if (cellContent) {
+        return false;
+      }
+
+      if (this.getters.getFilter(sheetId, col, row)) {
+        return false;
+      }
+
+      if (this.getters.isInMerge(sheetId, col, row)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  // ---------------------------------------------------------------------------
+  // Import/Export
+  // ---------------------------------------------------------------------------
+
+  import(data: WorkbookData) {
+    for (const sheet of data.sheets) {
+      for (const filterTableData of sheet.filterTables || []) {
+        const table = this.createFilterTable(toZone(filterTableData.range));
+        this.history.update("tables", sheet.id, table.id, table);
+      }
+    }
+  }
+
+  export(data: WorkbookData) {
+    for (const sheet of data.sheets) {
+      for (const filterTable of this.getFilterTables(sheet.id)) {
+        sheet.filterTables.push({
+          range: zoneToXc(filterTable.zone),
+        });
```

Comment:
```suggestion
        sheet.filterTables.push(zoneToXc(filterTable.zone));
```
Any value to have this object with a `range` key?

Reply:
Since I remove the filtered values from core, not really.
But having `{range : "xc"}` instead of a  simple string also meansa bit less work/migration when/if in the future we add things to Tables (styles, tables name, ...). 
I can change if you want, but this doesn't really affect anything.

Reply:
I have mixed feeling.
On one hand you've got a point it's more future proof. On the other I like to keep things simple while we don't need it (YAGNI)
Neither Google nor Excel seem to have any options to filter table. And I don't think we will either
Hence I'd choose the simple version

Reply:
Excel definitevely have tons of options to add to tables. Nos so much for GSheet.
But I get your point, I'll change it.